### PR TITLE
fix: screen orientation

### PR DESCRIPTION
**Pull request by MounirDhahri**
Link: https://www.notion.so/artsy/Go-to-any-page-without-having-your-screen-orientation-locked-turn-around-the-phone-we-get-this-f-13dcab0764a0802b9646de4dbbd81fc9?pvs=4

### Description

This PR fixes an issue with the screen orientation after enabling the new navigation infra.

| After | Before |
|--------|--------|
| <Video src="https://github.com/user-attachments/assets/2f8edc17-749d-43e3-80b9-5d7af976e0c5" /> | <Video src="https://github.com/user-attachments/assets/ac463118-4f1d-4b05-9088-907b45295789" /> | 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- fix screen orientation on new navigation - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (fix screen orientation on new navigation - mounir - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3357b426629622bb2b469eac97d0abbe96b71f5c
</p>

File: src/app/Navigation/AuthenticatedRoutes/StackNavigator.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ export const registerScreen: React.FC<StackNavigatorScreenProps> = ({ name, modu
       options={{
         presentation: isModalScreen(module) ? "fullScreenModal" : "card",
         headerShown: isHeaderShown(module),
+        orientation: "portrait",
```
Comment (brainbicycle) [inline src/app/Navigation/AuthenticatedRoutes/StackNavigator.tsx:35]: I may be wrong but I think previously we had it locked to portrait for non-tablet devices but allowed rotation for tablets?

Comment (gkartalis) [reply src/app/Navigation/AuthenticatedRoutes/StackNavigator.tsx:35]: that does ring a bell, other than that it sounds as a reasonable change

Comment (MounirDhahri) [reply src/app/Navigation/AuthenticatedRoutes/StackNavigator.tsx:35]: Great catch. This is true!

**Review by brainbicycle — APPROVED** [review]
🙌

**Pull request merged**